### PR TITLE
boards: unified LED macros

### LINKED ISSUE #5028: boards: Unify LED_(.*)_(ON|OFF|TOGGLE)
See https://github.com/RIOT-OS/RIOT/issues/5027#issuecomment-194821145.

Basically every board choses the colours as they like to.

I'm not advocating however to remove or replace them. Maybe a unifying API macro alá `LED_ON(color)` that is the solution here.

### PR DESCRIPTION
**Pull request by haukepetersen**
fixes #5028 and solves #5027

Now all boards use the same (numeric) naming scheme for their on-board LEDs (`LED0_xx`, `LED1_xx`, ...). Also each boards defines as many LEDs as available - and none if there are none.

The test is changes in a way, that always all available LEDs are used, so the behavior is easy to understand.

Tested for:
- [x] airfy-beacon
- [x] arduino-due
- [x] arduino-mega2560
- [ ] avsextrem
- [ ] cc2538dk
- [ ] chronos
- [ ] ek-lm4f120xl
- [x] f4vi1
- [ ] fox
- [ ] frdm-k64f
- [x] iotlab-m3
- [ ] limifrog-v1
- [x] mbed_lpc1768
- [ ] msb-430
- [ ] msb-430h
- [ ] msba2
- [x] msbiot
- [ ] mulle
- [x] native
- [x] nrf51dongle
- [x] nrf52dk
- [ ] nrf6310
- [x] nucleo-f091
- [ ] nucleo-f103
- [ ] nucleo-f303
- [x] nucleo-f334
- [ ] nucleo-f401
- [x] nucleo-l1
- [ ] openmote-cc2538
- [x] pba-d-01-kw2x
- [x] pca10000
- [x] pca10005
- [ ] pttu
- [x] qemu-i386
- [ ] remote
- [ ] saml21-xpro
- [x] samr21-xpro
- [ ] slwstk6220a
- [x] spark-core
- [ ] stm32f0discovery
- [ ] stm32f3discovery
- [ ] stm32f4discovery
- [x] telosb
- [x] udoo
- [ ] weio
- [ ] wsn430-v1_3b
- [ ] wsn430-v1_4
- [x] yunjia-nrf51822
- [x] z1

### CODE REVIEW

**Comment by authmillenon** [comment]
Needs rebase.

**Comment by authmillenon** [comment]
Also: I pointed this already out to you in person, but this doesn't really solve the real problem I opened #5028 for. You still need to guard `LED1_ON` e.g. if you want to use it since there is no guarantee it is defined. Since your PR even removes some of the nil defines it even worsens the problem.

**Comment by authmillenon** [comment]
Buuut some intermediate header that defines the undefined LED macros might be the solution for that (which could be solved in a separate PR).

**Comment by authmillenon** [comment]
Using numbers instead of arbitrary color names is definitely a step in a better direction IMHO.

**Comment by OlegHahm** [comment]
> Also: I pointed this already out to you in person, but this doesn't really solve the real problem I opened #5028 for.

Then you should probably rephrase #5028. From the title and description this seems to solve exactly what you describe in #5028.

**Comment by authmillenon** [comment]
> Then you should probably rephrase #5028. From the title and description this seems to solve exactly 
what you describe in #5028.

Well the portability issue is more a secondary issue that was for me logically following from the naming issue. Will open a new issue for portability.

**Comment by haukepetersen** [comment]
how about a simple `led.h` file, that defines all LED0-LED7 macros like this:
```c
#include "board.h"

#ifndef LED0_xx
#define LED0_xx   // nothing to do
#endif
...
```

**Comment by authmillenon** [comment]
> This would be one hell of an ugly file, but should work

That's kind of a given with macros :P

**Comment by authmillenon** [comment]
But +1 for the idea in general.

**Comment by haukepetersen** [comment]
rebased and fixed typo

**Comment by OlegHahm** [comment]
ACK, please squash.

**Comment by haukepetersen** [comment]
rebased and squashed, waiting for our CI friend

**Comment by haukepetersen** [comment]
fixeds some more issues with calls to `LED_RED_xx`.

**Comment by haukepetersen** [comment]
damn, a bunch of unrelated doxygen warnings... Will fix them

File: tests/periph_spi/main.c [hunk 1]
```diff
@@ -131,11 +131,9 @@ void slave_on_cs(void *arg)
 {
     (void)arg;
 
-    LED_RED_ON;
```
Comment (authmillenon) [inline tests/periph_spi/main.c]: Shouldn't those be replaced rather than removed?

File: drivers/lm75a/lm75a-temp-sensor.c [hunk 2]
```diff
@@ -299,17 +299,13 @@ void lm75A_start_sensor_sampling(void (*handler)(void))
...
-        LED_RED_TOGGLE;
         printf("amb_temp = %3.3f\n", lm75A_get_ambient_temperature());
 
         if (my_alarm && (handler != NULL)) {
-            LED_GREEN_ON;
             handler();
             my_alarm = false;
         }
         xtimer_usleep(100000);
-        LED_RED_TOGGLE;
```
Comment (authmillenon) [inline drivers/lm75a/lm75a-temp-sensor.c]: dito on those.

**Comment by authmillenon** [comment]
Just two questions for clarification. Other than that, re-ACK.

**Comment by haukepetersen** [comment]
yepp, those two should not be there anymore, I thought I removed them, but apparently not...

**Comment by haukepetersen** [comment]
fixed two newline at EOF issues and added the missing `periph_conf.h` to the `qemu-i386` board

**Comment by haukepetersen** [comment]
got 2 acks and Murdock is finally happy (after ~10000 tries) -> go

**Comment by kaspar030** [comment]
> Murdock is finally happy (after ~10000 tries)

imagine 12 hrs of waiting each time. ;)

**Comment by haukepetersen** [comment]
...and the best-thing-that-happened-to-RIOT-this-year award goes to ... Murdock!

**Pull request merged**